tests/conftest.py

```diff
@@ -410,6 +417,10 @@ async def generate_test_database_connection_url(
             pass
         finally:
             await connection.close()
+    else:
+        raise ValueError(
+            f"Unknown scheme {scheme!r} parsed from database url {original_url!r}."
+        )
```

Comment:
Without this change, if the schema cannot be parsed pytest raises a `StopIteration` error since the async generator never yields and what you've done wrong is utterly inscrutable.